### PR TITLE
BugFix: vtctld panic with `enable_realtime_stats`

### LINKED ISSUE #10901: Bug Report: vtctld crashes with `enable_realtime_stats` flag
### Overview of the Issue

Starting vtctld with `enable_realtime_stats` flag causes it to crash with the following error message - 
```
panic: non-positive interval for NewTicker

goroutine 69 [running]:
time.NewTicker(0x323d070?)
        time/tick.go:24 +0x10f
vitess.io/vitess/go/vt/discovery.(*TopologyWatcher).Start.func1(0xc00020f2c0)
        vitess.io/vitess/go/vt/discovery/topology_watcher.go:130 +0x77
created by vitess.io/vitess/go/vt/discovery.(*TopologyWatcher).Start
        vitess.io/vitess/go/vt/discovery/topology_watcher.go:128 +0x75
```

### Reproduction Steps

Run vtctld with `enable_realtime_stats` flag

### Binary Version

```sh
main. Release-14.0 and before don't seem to be affected.
```

### Operating System and Environment details

```sh
All
```

### Log Fragments

_No response_

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR fixes the panic as described in https://github.com/vitessio/vitess/issues/10901.

The problem was traced back to https://github.com/vitessio/vitess/pull/10863. In this refactor of flags, we accidentally removed discovery flags for being registered for vtctld, which caused the default value of the flag not being used while starting the timer, causing it to panic.

## Related Issue(s)

Fixes #10901 

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/discovery/healthcheck.go [hunk 1]
```diff
@@ -312,6 +313,7 @@ func NewHealthCheck(ctx context.Context, retryDelay, healthCheckTimeout time.Dur
 	if cellsToWatch == "" {
 		cells = append(cells, localCell)
 	}
+	log.Errorf("Refresh Interval - %v", refreshInterval)
```
Comment (ajm188) [inline go/vt/discovery/healthcheck.go:316]: nit: we should delete this

File: go/vt/discovery/healthcheck.go [hunk 2]
```diff
@@ -149,6 +149,7 @@ func ParseTabletURLTemplateFromFlag() {
 func init() {
 	servenv.OnParseFor("vtgate", registerDiscoveryFlags)
 	servenv.OnParseFor("vtcombo", registerDiscoveryFlags)
+	servenv.OnParseFor("vtctld", registerDiscoveryFlags)
```
Comment (ajm188) [inline go/vt/discovery/healthcheck.go:152]: do we need `vtctl` as well, or _just_ `vtctld` (i assume just `vtctld`, but want to make sure we've thought about it)

**Review by ajm188 — CHANGES_REQUESTED** [review]

File: go/vt/discovery/healthcheck.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/discovery/healthcheck.go:316]: Oh yes, for sure. Sorry I was using that to debug what was going on!

File: go/vt/discovery/healthcheck.go (hunk 2 above)
Comment (GuptaManan100) [reply go/vt/discovery/healthcheck.go:152]: I don't think `vtctl` requires it, but there is something in `go/vt/throttler/demo/throttler_demo.go` which is also using the healthcheck. I don't if we need to add it 🤷‍♂️

Comment (GuptaManan100) [reply go/vt/discovery/healthcheck.go:152]: `Vttablet` too

**Review by deepthi — COMMENTED** [review]
Thank you for catching this.
Here's what I think we need to do
- Add the flags for `vtctld` (this PR already covers this) + a test for `vtctld`
- `throttler_demo` is a separate binary. Instead of adding flags to it, we should make it use the defaults
- `vttablet`'s tx_throttler should use default values (similar to how it already calls `NewHealthCheck`)
- `vtexplain` uses `FakeHealthCheck` so I don't think it needs to change
- I'm unsure about `vtctl`, we'll need to dig deeper to figure this out.

**Comment by deepthi** [comment]
When I did #10863, @ajm188 and I went back and forth about setting default values along with the var declaration versus setting them in tests. I'm glad that the way we did it exposed this issue, but it is time to consider setting defaults for the `throttler_demo` and `vttablet` usages to work.

File: go/vt/discovery/healthcheck.go (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/discovery/healthcheck.go]: Fixed

**Review by GuptaManan100 — COMMENTED** [review]
I have added the test and added the defaults for the values.

**Review by deepthi — APPROVED** [review]

**Review by ajm188 — APPROVED** [review]

**Pull request merged**